cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/MaintenancePlanSerdeTest.java

```diff
@@ -0,0 +1,77 @@
+  }
+
+  @Test
+  public void testAddBrokerPlanSerde() {
+    MaintenancePlanSerde maintenancePlanSerde = new MaintenancePlanSerde();
+    AddBrokerPlan addBrokerPlanA = new AddBrokerPlan(System.currentTimeMillis(), TEST_BROKER_ID, Collections.singleton(TEST_BROKER_ID));
+    byte[] serializedJson = maintenancePlanSerde.serialize(TOPIC_NAME, addBrokerPlanA);
+    AddBrokerPlan addBrokerPlanB = (AddBrokerPlan) maintenancePlanSerde.deserialize(TOPIC_NAME, serializedJson);
+    Assert.assertNotNull(addBrokerPlanB);
+    Assert.assertEquals(addBrokerPlanA.getCrc(), addBrokerPlanB.getCrc());
+  }
+
+  @Test
+  public void testRemoveBrokerPlanSerde() {
+    MaintenancePlanSerde maintenancePlanSerde = new MaintenancePlanSerde();
+    RemoveBrokerPlan removeBrokerPlanA = new RemoveBrokerPlan(System.currentTimeMillis(), TEST_BROKER_ID, Collections.singleton(TEST_BROKER_ID));
+    byte[] serializedJson = maintenancePlanSerde.serialize(TOPIC_NAME, removeBrokerPlanA);
+    RemoveBrokerPlan removeBrokerPlanB = (RemoveBrokerPlan) maintenancePlanSerde.deserialize(TOPIC_NAME, serializedJson);
+    Assert.assertNotNull(removeBrokerPlanB);
+    Assert.assertEquals(removeBrokerPlanA.getCrc(), removeBrokerPlanB.getCrc());
+  }
+
+  @Test
+  public void testFixOfflineReplicasPlanSerde() {
+    MaintenancePlanSerde maintenancePlanSerde = new MaintenancePlanSerde();
+    FixOfflineReplicasPlan fixOfflineReplicasPlanA = new FixOfflineReplicasPlan(System.currentTimeMillis(), TEST_BROKER_ID);
+    byte[] serializedJson = maintenancePlanSerde.serialize(TOPIC_NAME, fixOfflineReplicasPlanA);
+    FixOfflineReplicasPlan fixOfflineReplicasPlanB = (FixOfflineReplicasPlan) maintenancePlanSerde.deserialize(TOPIC_NAME, serializedJson);
+    Assert.assertNotNull(fixOfflineReplicasPlanB);
+    Assert.assertEquals(fixOfflineReplicasPlanA.getCrc(), fixOfflineReplicasPlanB.getCrc());
+  }
+
+  @Test
+  public void testRebalancePlanSerde() {
+    MaintenancePlanSerde maintenancePlanSerde = new MaintenancePlanSerde();
+    RebalancePlan rebalancePlanA = new RebalancePlan(System.currentTimeMillis(), TEST_BROKER_ID);
+    byte[] serializedJson = maintenancePlanSerde.serialize(TOPIC_NAME, rebalancePlanA);
+    RebalancePlan rebalancePlanB = (RebalancePlan) maintenancePlanSerde.deserialize(TOPIC_NAME, serializedJson);
+    Assert.assertNotNull(rebalancePlanB);
+    Assert.assertEquals(rebalancePlanA.getCrc(), rebalancePlanB.getCrc());
+  }
+
+  @Test
+  public void testTopicReplicationFactorPlanSerde() {
+    MaintenancePlanSerde maintenancePlanSerde = new MaintenancePlanSerde();
+    TopicReplicationFactorPlan topicReplicationFactorPlanA =
+        new TopicReplicationFactorPlan(System.currentTimeMillis(), TEST_BROKER_ID, Collections.singletonMap((short) 42, TOPIC_NAME));
+    byte[] serializedJson = maintenancePlanSerde.serialize(TOPIC_NAME, topicReplicationFactorPlanA);
+    TopicReplicationFactorPlan topicReplicationFactorPlanB =
```

Comment:
Similar to https://github.com/linkedin/cruise-control/pull/1515/files#r613605637,
Can we
1. Make `TopicReplicationFactorPlan#_topicRegexWithRFUpdate` a `SortedMap`, and
2. Add a test in this class that verifies the TopicReplicationFactorPlan with multiple entries?

Reply:
good point. done.